docs/api/migration-guides/qiskit-1.0.mdx

```diff
@@ -0,0 +1,659 @@
+If the problem persists, at least one of the packages you want likely does not support Qiskit 1.0 yet, and `pip` is finding an old version that it believes will work because it doesn't know about the `qiskit>=1`/`qiskit-terra` conflict.
+
+Instead, use the `pipdeptree` commands to identify which dependencies do not yet support Qiskit 1.0.
+Exclude any packages that do not yet support Qiskit 1.0 when constructing a Qiskit 1.0 environment, or continue to use a prior version of Qiskit.
+See [Create the new environment](#creating-a-venv) for instructions.
+
+<Admonition type="note">
+The example in this section was generated before Qiskit 1.0 was released.
+
+The "old" distribution in question (`qiskit-dynamics`) was behaving correctly; it was not known to support Qiskit 1.0 yet, so it marked that in its requirements.
+It's not possible to backdate requirements changes to previously released versions, and `pip` will search arbitrarily far back to locate something that works when building an environment.
+</Admonition>
+
+
+
+<span id="debug-venv-for-0.45"></span>
+#### Create a working environment for Qiskit 0.45 or 0.46
+
+If you have a broken environment after trying to install Qiskit 0.45 or 0.46, the most likely situation is that `pip` installed Qiskit 1.0 because it tried to pick the latest versions of packages, even though it was not required.
+The easiest way to fix this is to create a new virtual environment, then run a single `pip install` command that has all the packages you need, plus an explicit `'qiskit<1'` entry.
+If `pip` successfully resolves this dependency graph, you should have a working virtual environment.
+If at least one distribution requires Qiskit 1.0 or greater, `pip` should give you an error message explaining this, which looks like the one in [the section on failed resolutions](#pip-resolution-impossible).
+
+You can also use the `pipdeptree` commands listed in [Create a working environment for Qiskit 1.0](#debug-venv-for-1.0) from within the broken environment to determine which distributions have an explicit requirement on `qiskit>=1`.
+
+
+#### I'm a developer, my environments are definitely right, and I'm still getting the error
+
+First: you must be *absolutely* certain that your environments are correct.
+The test that Qiskit uses to determine a broken environment is quite robust; specifically, it queries `importlib.metadata` for distribution information on installed packages and checks the version numbers returned.
+The Qiskit 1.0 side of the test also checks for sentinel files that were present in old Qiskit versions and not Qiskit 1.0.
+
+If you are a Qiskit developer, it's possible that you have old `qiskit.egg-info` or `qiskit-terra.egg-info` (or `*.dist-info`) directories present on your meta path (see `sys.meta_path`), left over from old editable installations.
+In particular, check your working directory for any `*.egg-info` and `*.dist-info` directories.
```

Comment:
I don't know if anyone is reading this far down in the guide, but I think the problem could also be `__pycache__` directories if you use `pip install -e`. You could suggest `git clean -fxd` after the user checks that there are no important uncommitted files.

Reply:
`__pycache__` shouldn't cause problems here, because the first `importlib` search is for distributions not packages, and then the second is a search for "is `qiskit.tools` a loadable module _with a backing source file_?".  It's not enough for the `qiskit/tools` directory to be on the path, even if `qiskit/tools/__pycache__/__init__.*.pyc` exists, because we test that the `ModuleSpec` has a proper location for the file (`qiskit/tools.py` or `qiskit/tools/__init__.py`) to verify it's not an empty namespace package left over from a dirty source tree.

I'm hesitant to suggest `git clean` for general use because it's too easy to accidentally take out local changes, or clean out all your cached tox environments, etc.

Reply:
My comment was misleading. When I said "this problem," I was thinking of the problem of left over files repo that had been installed in editable mode rather than the specific `import qiskit` problem with left over namespace package metadata.

I didn't fully test it but somehow there were some code removals that didn't trigger import errors for me with my editable install, and I had assumed the imports were still working because that is how things worked with Python 2 and the .pyc files left over in directories that had had the .py files removed. Maybe it was actually the info directories or something else. I didn't try to recreate the problem after doing the clean.

That particular case was coming up for me with other packages that set up Qiskit plugins in the package metadata (specifically qiskit-ibm-provider's transpiler pass plugin that imported from `qiskit.extensions`). It kept working in my editable install but then failed in the clean CI environment. It is a surprising case because I was not using the plugin or anything else from qiskit-ibm-provider but just having the distribution with the plugin installed is enough to trigger an error when trying to do a transpile. I was wondering if there was anything to mention related to that in the troubleshooting section, but I think ultimately it is a consequence of having a package that is not compatible with Qiskit 1.0 installed, so that is already covered.

Reply:
Could you send me a set of repro steps, and I'll look into either making the import check smarter, or adding an extra bit to the troubleshooting section in this guide in a day or so?